Lib/test/test_buffer.py

```diff
@@ -4412,6 +4417,12 @@ def test_issue_7385(self):
         x = ndarray([1,2,3], shape=[3], flags=ND_GETBUF_FAIL)
         self.assertRaises(BufferError, memoryview, x)
 
+    @support.cpython_only
+    def test_pybuffer_size_from_format(self):
+        nitems = randrange(1, 30)
```

Comment:
I would prefer to avoid using random in such test. iter_format() returns 91 items: it sounds overkill to me. IMHO tTesting 3 formats should but enough, like:

```
for fmt in ('', 'ii', '3s'):
```